blackbox-test-inject/src/main/java/org/example/myapp/conditional/BirdFactory.java

```diff
@@ -0,0 +1,27 @@
+package org.example.myapp.conditional;
+
+import org.example.myapp.conditional.Bird.BlueJay;
+import org.example.myapp.conditional.Bird.Cassowary;
+
+import io.avaje.inject.Bean;
+import io.avaje.inject.Factory;
+import io.avaje.inject.RequiresProperty;
+import io.avaje.inject.Secondary;
+
+@Factory
+@RequiresProperty(value = "factory", equalTo = "bird")
+public class BirdFactory {
+
+  @Bean
+  @NoKiwi
+  public BlueJay jay() {
```

Comment:
Rob, check that the RequiresProperty is 'inherited' for all the methods of the factory.